File: android/src/main/java/com/hlabexmaples/easyapi/data/webservice/ApiRepository.java
package com.hlabexmaples.easyapi.data.webservice;

import com.hlabexmaples.easyapi.data.easyapi.main.model.Base;
import com.hlabexmaples.easyapi.data.models.Envelop;
import com.hlabexmaples.easyapi.data.models.User;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Retrofit callback methods.
 */

public interface ApiRepository {

    @GET(Constants.WS_METHOD_USERS)
    Call<Envelop<List<User>>> fetchUsers(@Query(Constants.PARAM_PAGE) String page);

    @GET(Constants.WS_METHOD_USERS)
    Observable<Envelop<List<User>>> fetchUsersWithRx(@Query(Constants.PARAM_PAGE) String page);

    /*Example with Base class*/
    @GET(Constants.WS_METHOD_USERS)
    Observable<Base<List<User>>> fetchUsersBase(@Query(Constants.PARAM_PAGE) String page);

}

File: kotlin/src/main/java/com/hlabexamples/easyapi/kotlin/data/webservice/ApiRepository.kt
package com.hlabexamples.easyapi.kotlin.data.webservice

import com.hlabexamples.easyapi.kotlin.data.easyapi.main.Base
import com.hlabexamples.easyapi.kotlin.data.models.Envelop
import com.hlabexamples.easyapi.kotlin.data.models.User
import io.reactivex.Observable
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Retrofit callback methods.
 */

interface ApiRepository {

    @GET(Constants.WS_METHOD_USERS)
    fun fetchUsers(@Query(Constants.PARAM_PAGE) page: String): Call<Envelop<List<User>>>

    @GET(Constants.WS_METHOD_USERS)
    fun fetchUsersWithRx(@Query(Constants.PARAM_PAGE) page: String): Observable<Envelop<List<User>>>

    /*Example with Base class*/
    @GET(Constants.WS_METHOD_USERS)
    fun fetchUsersBase(@Query(Constants.PARAM_PAGE) page: String): Observable<Base<List<User>>>

}
